From a335e9d445f15ff4ab0bb5982db8088eb47f6be2 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 10:36:24 +0800
Subject: [PATCH 244/301] AOSCOS: drm: ttm: introduce struct delayed_work
 member dwork to struct ttm_device

Commit 9bff18d13473 ("drm/ttm: use per BO cleanup workers") revised the
member struct `wq' for struct `ttm_device' from type `struct delayed_work'
to `struct workqueue_struct'.

This broke the ttm_bo_{un,}lock_delayed_workqueue() functions introduced
in "AOSCOS: Revert "drm/ttm: remove ttm_bo_(un)lock_delayed_workqueue"",
as they expect a `struct delayed_work' type as the first argument - which
is now revised as `struct workqueue_struct'.

Introduce a `struct delayed_work dwork' member to fix build (but would it
still work as expected?).

Fixes: "AOSCOS: Revert "drm/ttm: remove ttm_bo_(un)lock_delayed_workqueue""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/ttm/ttm_bo.c | 4 ++--
 include/drm/ttm/ttm_device.h | 5 +++++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/ttm/ttm_bo.c b/drivers/gpu/drm/ttm/ttm_bo.c
index 2529f8ee0321..1c049a1442a0 100644
--- a/drivers/gpu/drm/ttm/ttm_bo.c
+++ b/drivers/gpu/drm/ttm/ttm_bo.c
@@ -331,14 +331,14 @@ EXPORT_SYMBOL(ttm_bo_put);
 
 int ttm_bo_lock_delayed_workqueue(struct ttm_device *bdev)
 {
-	return cancel_delayed_work_sync(&bdev->wq);
+	return cancel_delayed_work_sync(&bdev->dwork);
 }
 EXPORT_SYMBOL(ttm_bo_lock_delayed_workqueue);
 
 void ttm_bo_unlock_delayed_workqueue(struct ttm_device *bdev, int resched)
 {
 	if (resched)
-		schedule_delayed_work(&bdev->wq,
+		schedule_delayed_work(&bdev->dwork,
 				      ((HZ / 100) < 1) ? 1 : HZ / 100);
 }
 EXPORT_SYMBOL(ttm_bo_unlock_delayed_workqueue);
diff --git a/include/drm/ttm/ttm_device.h b/include/drm/ttm/ttm_device.h
index 39b8636b1845..07594b72de63 100644
--- a/include/drm/ttm/ttm_device.h
+++ b/include/drm/ttm/ttm_device.h
@@ -267,6 +267,11 @@ struct ttm_device {
 	 * @wq: Work queue structure for the delayed delete workqueue.
 	 */
 	struct workqueue_struct *wq;
+
+	/**
+	 * @dwork: Delayed work in the workqueue.
+	 */
+	struct delayed_work dwork;
 };
 
 int ttm_global_swapout(struct ttm_operation_ctx *ctx, gfp_t gfp_flags);
-- 
2.48.1

